libraries/botbuilder-ai/src/lgResolver.ts

```diff
@@ -0,0 +1,907 @@
+/**
+ * @module botbuilder-ai
+ *
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import {
+	Activity,
+	CardAction,
+	CardFactory,
+	MediaCard,
+	OAuthCard,
+	ReceiptCard,
+	ReceiptItem,
+	SigninCard,
+	ThumbnailCard,
+} from 'botbuilder';
+import {
+	flatten,
+	isBoolean,
+	isDate,
+	isEmpty,
+	isInteger,
+	isNil,
+	isNumber,
+	isString,
+	replace,
+} from 'lodash';
+import * as request from 'request-promise-native';
+
+//  ######################################### EXPORTED API #########################################
+
+export type PrimitiveType = string | number | boolean | Date;
+
+/**
+ * Description of a language generation application used for initializing a LanguageGenerationResolver.
+ */
+export interface LanguageGenerationApplication {
+	/* Language generation application id */
+	applicationId: string;
+
+	/** (Optional) Azure region */
+	azureRegion?: string;
+
+	/* Congnetive service authorization key */
+	endpointKey: string;
+}
+
+/**
+ * Options for LanguageGenerationResolver.
+ */
+export interface LanguageGenerationOptions {}
+
+/**
+ * Component used to extract template references from the activity object, resolve them using the language generation service
+ * and inject them back in the activity.
+ */
+export class LanguageGenerationResolver {
+	private lgApi: LGAPI;
+	constructor(
+		private application: LanguageGenerationApplication,
+		private options: LanguageGenerationOptions
+	) {
+		this.validateApplicationInputs();
+		this.lgApi = new LGAPI(application, options);
+	}
+
+	/**
+	 * Extracts template references from the activity object, resolves them using the language generation service
+	 * and injects them back in the activity
+	 *
+	 * @param activity botbuilder activity
+	 * @param [entities] map object that contains slots accepted by the LG service
+	 * @returns
+	 */
+	public async resolve(activity: Activity, entities?: Map<string, PrimitiveType>): Promise<void> {
```

Comment:
>Map<string, PrimitiveType> [](start = 53, length = 26)

Instead of Map I'd just make this a JSON object so { [name: string]: PrimitiveType; }

That will be easier for the caller versus having to new up and populate an actual Map